go/test/endtoend/prepare-statement/stmt_methods_test.go

```diff
@@ -0,0 +1,170 @@
+	deleteRecord(t, dbo)
+
+	// testing recontion and deleted data validation
+	reconnectAndTest(t)
+}
+
+// testcount validates inserted rows count with expected count.
+func testcount(t *testing.T, dbo *sql.DB, except int) {
+	r, err := dbo.Query("SELECT count(1) FROM " + tableName)
+	require.Nil(t, err)
+
+	// read output of query in var i
+	r.Next()
+	var i int
+	err = r.Scan(&i)
+	assert.Nil(t, err)
+	assert.Equal(t, except, i)
+}
+
+// TestAutoIncColumns test insertion of row without passing
+// the value of auto increment columns (here it is id).
+func TestAutoIncColumns(t *testing.T) {
+	dbo := GetORM(t)
+	defer dbo.Close()
+	// insert a row without id
+	insertStmt := "INSERT INTO " + tableName + ` (
+		msg,keyspace_id,tinyint_unsigned,bool_signed,smallint_unsigned,
+		mediumint_unsigned,int_unsigned,float_unsigned,double_unsigned,
+		decimal_unsigned,t_date,t_datetime,t_time,t_timestamp,c8,c16,c24,
+		c32,c40,c48,c56,c63,c64,json_col,text_col,data) VALUES (?,  ?,  ?,  ?,  ?, ?,
+		  ?,  ?,  ?,  ?,  ?,  ?,  ?,  ?,  ?,  ?,  ?, ?,  ?,  ?,  ?,  ?,  ?, ?, ?, ?);`
+	insertValue := []interface{}{
+		"21", 0,
+		127, 1, 32767, 8388607, 2147483647, 2.55, 64.9, 55.5,
+		time.Date(2009, 5, 5, 0, 0, 0, 0, time.UTC),
+		time.Date(2009, 5, 5, 0, 0, 0, 0, time.UTC),
+		time.Now(),
+		time.Date(2009, 5, 5, 0, 0, 0, 0, time.UTC),
+		1, 1, 1, 1, 1, 1, 1, 1, 1, jsonExample, fake.DomainName(), fake.Paragraph(),
+	}
+
+	exec(t, dbo, insertStmt, insertValue...)
+}
+
+// deleteRecord test deletion opeation corresponds to the testingID.
+func deleteRecord(t *testing.T, dbo *sql.DB) {
+	// delete the record with id 1
+	exec(t, dbo, "DELETE FROM "+tableName+" WHERE id = ?;", testingID)
+
+	data := selectWhere(t, dbo, "id = ?", testingID)
+	assert.Equal(t, 0, len(data))
+
+}
+
+// updateRecord test update opeation corresponds to the testingID.
+func updateRecord(t *testing.T, dbo *sql.DB) {
+	// update the record with id 1
+	updateData := "new data value"
+	updateTextCol := "new text col value"
+	updateQuery := "update " + tableName + " set data = ? , text_col = ? where id = ?;"
+
+	exec(t, dbo, updateQuery, updateData, updateTextCol, testingID)
+
+	// validate the updated value
+	// validate row count
+	data := selectWhere(t, dbo, "id = ?", testingID)
+	assert.Equal(t, 1, len(data))
+
+	// validate value of msg column in data
+	assert.Equal(t, updateData, data[0].Data)
+	assert.Equal(t, updateTextCol, data[0].TextCol)
+
+}
+
+// reconnectAndTest creates new connection with database and validate.
+func reconnectAndTest(t *testing.T) {
+	// reconnect and try to select the record with id 1
+	dbo := GetORM(t)
+	defer dbo.Close()
+	data := selectWhere(t, dbo, "id = ?", testingID)
+	assert.Equal(t, 0, len(data))
+
+}
+
+// TestWrongTableName query database using invalid
+// tablename and validate error.
+func TestWrongTableName(t *testing.T) {
+	dbo := GetORM(t)
+	defer dbo.Close()
+	execWithIgnore(t, dbo, []uint16{1105}, "select * from teseting_table;")
```

Comment:
The test will also pass if there is no error. We need to assert that the expected error has actually occurred.